File: 001/python/main.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        cache = {}
        for i, num in enumerate(nums):
            r = target - num
            if r in cache:
                return cache[r], i
            cache[num] = i
        return None

File: 001/cpp/main.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::map<int, int> cache;
        int size = nums.size();
        for (int i = 0; i < size; i++) {
            int t = target - nums[i];
            if (cache.find(t) != cache.end()) {
                return {cache[t], i};
            }
            cache[nums[i]] = i;
        }
        return {};
    }
};
